syntax="proto3";

message Product {
    string id = 1;
    string name = 2;
    string category = 3;
    int32 price = 4;
    int32 quantity = 5;
}

message Order {
    string product_id = 1;
    string order_id = 2;
    string product_name = 3;
    string product_category = 4;
    int32 quantity = 5;
    int32 product_price = 6;
    int32 totle_price = 7;
} 

message UpdateProduct {
    string id = 1;
    optional string name = 2;
    optional string category = 3;
    optional int32 price = 4;
    optional int32 quantity = 5;
}

message IncrementProductItem {
    string id = 1;
    int32 add_product = 2;
}